src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py

```diff
@@ -15,3 +15,16 @@ def load_arguments(self, _):
         c.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='start date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
         c.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
         c.argument('billing_period_name', options_list=['--billing-period-name', '-p'], help='name of a specific billing period to get the usage details that associate with')
+    
+    with self.argument_context('consumption reservations summaries list') as rs:
+        rs.argument('reservationorderid', options_list=['--reservationorderid', '-roid'],help='Reservation order id')
+        rs.argument('reservationid', options_list=['--reservationid', '-rid'], help='Reservation id')
+        rs.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='start date (in UTC Y-m-d) of the reservation summaries. Only needed for daily grain and both start date and end date need to be supplied or neither')
+        rs.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the reservation summaries. Only needed for daily grain and both start date and end date need to be supplied or neither')
+        rs.argument('grain', options_list=['--grain', '-g'], type=str,help='Reservation summaries grain. Possible values are daily or monthly')
+    
+    with self.argument_context('consumption reservations details list') as rd:
+        rd.argument('reservationorderid', options_list=['--reservationorderid', '-roid'],help='Reservation order id')
+        rd.argument('reservationid', options_list=['--reservationid', '-rid'], help='Reservation id')
+        rd.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='start date (in UTC Y-m-d) of the reservation summaries. Only needed for daily grain and both start date and end date need to be supplied or neither')
+        rd.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the reservation summaries. Only needed for daily grain and both start date and end date need to be supplied or neither')        
```

Comment:
Define the shared params used in both commands by putting them in the same context as `consumption reservations`. This will allow the settings to apply for all commands underneath that group.

Reply:
As discussed , for now i am keeping  separate so that changes don't  affect each other